src/Components/test/E2ETest/Tests/ErrorNotificationServerSideTest.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using BasicTestApp;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.Components.E2ETest.ServerExecutionTests;
+using Microsoft.AspNetCore.E2ETesting;
+using OpenQA.Selenium;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.Tests
+{
+    public class ErrorNotificationServerSideTest : ServerTestBase<ToggleExecutionModeServerFixture<Program>>
+    {
+        public ErrorNotificationServerSideTest(
+            BrowserFixture browserFixture,
+            ToggleExecutionModeServerFixture<Program> serverFixture,
+            ITestOutputHelper output)
+            : base(browserFixture, serverFixture.WithServerExecution(), output)
+        {
+        }
+
+        protected override void InitializeAsyncCore()
+        {
+            // On WebAssembly, page reloads are expensive so skip if possible
+            Navigate(ServerPathBase, noReload: _serverFixture.ExecutionMode == ExecutionMode.Server);
+            Browser.MountTestComponent<ErrorComponent>();
+            Browser.Exists(By.Id("error-ui"));
+            Browser.Exists(By.TagName("button"));
+        }
+
+        [Fact]
+        public void ShowsErrorNotificationServerSide_OnError()
+        {
+            var errorUi = Browser.FindElement(By.Id("error-ui"));
+            Assert.Equal("none", errorUi.GetCssValue("display"));
+
+            var causeErrorButton = Browser.FindElement(By.TagName("button"));
+            causeErrorButton.Click();
+
+            Browser.Exists(By.CssSelector("#error-ui[style='display: block;']"));
+
+            var reload = Browser.FindElement(By.ClassName("reload"));
+            reload.Click();
+
+            Browser.MountTestComponent<ErrorComponent>();
```

Comment:
I think there is potential for this test to be flaky from this point onwards. On line 46 you're making the browser reload, but not doing anything to wait until it has reloaded. So Selenium might charge ahead and, before the old page was unloaded, start capturing references to things in the old page (e.g., `causeErrorButton`) then when it does reload those references will be stale and it will throw.

I'd recommend splitting this into two tests:

 * One that checks the error message is displayed and that "reload" does cause the page to be reloaded. To assert the latter, you could just observe that some element from `ErrorComponent.razor` ceases to be findable in the page (e.g., that trying to find something with its CSS class returns an empty set).
 * One that checks the error message is displayed and that "dismiss" does cause it to become hidden.

Neither of these depends on waiting until the reload has completed, so shouldn't suffer the same issue.